pkg/labels/labels.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2019 the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package labels
+
+import (
+	"crypto/sha256"
+	"fmt"
+)
+
+// maxLabelLength stores the max label length value supported by kubernetes.
+const maxLabelLength = 63
+
+// GetValidLabelName converts an input string to valid kubernetes label string in accordance to rfc1035 DNS Label spec
+// (https://github.com/kubernetes/community/blob/master/contributors/design-proposals/architecture/identifiers.md)
+func GetValidLabelName(label string) string {
```

Comment:
I would advocate for changing this function name to `GetValidName` and changing the import alias in other files to `labels`. In use, it would then read as `labels.GetValidName(input)`, which has less repetition.

Reply:
Yes, I think it a good idea. I can update the function name to GetValidName. As for the alias, I wanted to use "labels", however, it collides with "k8s.io/apimachinery/pkg/labels". Can we use "label" instead?

Reply:
Ah, good point. We have a convention for these conflicts that's not yet made its way into the contributing docs (See #1387).

I think `velerolabels` would make sense with the convention.